src/java.base/share/classes/sun/security/provider/HSS.java

```diff
@@ -0,0 +1,835 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package sun.security.provider;
+
+import sun.security.util.*;
+import sun.security.x509.AlgorithmId;
+import sun.security.x509.X509Key;
+
+import java.io.*;
+import java.security.*;
+import java.security.spec.*;
+import java.util.Arrays;
+import java.util.Objects;
+
+public class HSS extends SignatureSpi {
+    private HSSPublicKey pubKey;
+    private ByteArrayOutputStream messageStream;
+
+    @Deprecated
+    protected void engineSetParameter(String param, Object value) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Deprecated
+    protected AlgorithmParameters engineGetParameter(String param) {
+        throw new UnsupportedOperationException();
+    }
+
+    protected void engineInitSign(PrivateKey privateKey) throws InvalidKeyException {
+        throw new InvalidKeyException("Signing is not supported");
+    }
+
+    protected byte[] engineSign() throws SignatureException {
```

Comment:
This API is not defined to throw UOE. Suggest throwing `SignatureException` instead with a message "Signing is not supported".

Though, technically this method should never be called, as `Signature.sign()` will throw a `SignatureException` because it has not been initialized for signing.